### PR TITLE
Fixes the issues of valid pack name

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33909

## Related Pull Requests
https://github.com/demisto/content/pull/11941

## Description
Because we are adding new rules to the package name there are packages that have failed in the build.
We need to update them in the content repo.
For some of the files we changed the name, and for the others we are simply adding an ignore to this error, As you can see in this PR https://github.com/demisto/content/pull/11941.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 1]
```diff
@@ -225,10 +243,12 @@ def _is_pack_meta_file_structure_valid(self):
...
 
             # check validity of pack metadata mandatory fields
-            name_field = metadata.get(PACK_METADATA_NAME, '').lower()
-            if not name_field or 'fill mandatory field' in name_field:
+            name_field = metadata.get(PACK_METADATA_NAME, '')
+            if not name_field:
                 if self._add_error(Errors.pack_metadata_name_not_valid(), self.pack_meta_file):
                     return False
+            else:
+                return self.valid_pack_name(name_field)
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:251]: remove the `return`.
it prevents the rest of the function's validation

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 2]
```diff
@@ -34,6 +34,9 @@
 ISO_TIMESTAMP_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
 ALLOWED_CERTIFICATION_VALUES = ['certified', 'verified']
 SUPPORT_TYPES = ['community', 'xsoar'] + SUPPORTED_CONTRIBUTORS_LIST
+INCORRECT_PREFIX_PACK_NAME = ["Playbook", "Integration", "Script"]
```
[the remark this answers is not in the document]
Comment (MosheEichler) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:37]: The test should fail if one of the words is the suffix of the pack name?

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 3]
```diff
@@ -34,6 +34,9 @@
...
+PATTERN = '[^a-zA-Z]pack[^a-z]|^pack$|^pack[^a-z]|[^a-zA-Z]pack$|[^A-Z]PACK[^A-Z]|^PACK$|^PACK[^A-Z]|[^A-Z]PACK$|' \
```
[the remark this answers is not in the document]
Comment (MosheEichler) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:38]: I thing we have to be case sensitive because something like this **SomethingPackSomething** have to fail and this **SomethingPackageSomething** should not fail

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -1125,7 +1125,19 @@ def missing_field_iin_pack_metadata(pack_meta_file, missing_fields):
 
     @staticmethod
     @error_code_decorator
-    def pack_metadata_name_not_valid():
+    def pack_metadata_name_not_valid(reason=None):
```
[the remark this answers is not in the document]
Comment (roysagi) [reply demisto_sdk/commands/common/errors.py:1161]: Using the same error code means that if you ignore this error for one reason - you ignore it for all reasons - the question is, is this what we want?
Also, why are we using an existing error in the first place? shouldn't we create a new one for this?

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:37]: fail by `contains`.
pack names should not contain those words

File: demisto_sdk/commands/common/errors.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply demisto_sdk/commands/common/errors.py:1161]: @MosheEichler 
please use a dedicated error for this validation

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py (hunk 3 above)
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:38]: a pattern of what?
improve the constant name.

**Review by yaakovi — APPROVED** [review]

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 5]
```diff
@@ -226,10 +248,12 @@ def _is_pack_meta_file_structure_valid(self):
                     return False
 
             # check validity of pack metadata mandatory fields
-            name_field = metadata.get(PACK_METADATA_NAME, '').lower()
-            if not name_field or 'fill mandatory field' in name_field:
```
Comment (reutshal) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:230]: Why did you remove the second condition?

**Review by reutshal — COMMENTED** [review]
Please fix changelog conflicts

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py (hunk 5 above)
Comment (MosheEichler) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:230]: Sorry. it's was my mistake, I thought it was unnecessary
